### PR TITLE
fix `mul_out` CUDA config for COO tensors

### LINKED ISSUE #79914: mul_out_sparse_cuda creates invalid CUDA configuration arguments
### 🐛 Describe the bug

To reproduce the issue

```
import torch

a = torch.tensor([[0., 1]]).cuda().to_sparse().requires_grad_(True)
b = torch.tensor([[0., 1]]).cuda().to_sparse().requires_grad_(True)

y = a * b
torch.sparse.sum(y).backward()
```

Note that this yields values of 0 for grid.x and block.x within [mul_out_sparse_cuda](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu#L509) while launching `sparseElementwiseKernelScalar`.

I suspect that this is a long-standing issue that we haven't discovered or haven't fixed.

### Versions

This was run on a from-source build on revision e675c33c51ece9deb5f5cf81b6df9121c928709a (recent master).

cc @nikitaved @pearu @cpuhrsch @amjames

### PR DESCRIPTION
**Pull request by nikitaved**
Fixes https://github.com/pytorch/pytorch/issues/79914

### CODE REVIEW

File: aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu [hunk 1]
```diff
@@ -495,7 +495,7 @@ SparseTensor& mul_out_sparse_cuda(const SparseTensor& t_, const SparseTensor& sr
 
   Tensor r_values_ = new_values_with_size_of(t_values_, max_nnz).zero_();
 
-  int64_t valueSize = t_values_.stride(0);
+  int64_t valueSize = t_values_.size(0);
```
Comment (nikitaved) [inline aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu:498]: So, basically, we fail with 0-dim tensors. I guess now we also get a speed-up for free, because before the block size was of size just 1 with non-scalar index Tensor :)

Comment (nikitaved) [reply aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu:498]: `valueSize` is solely used for deciding the grid/block size, so it should be still correct provided that indices are handled correctly.

Comment (nikitaved) [reply aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu:498]: It was a choice between 1 and 4 only.

Comment (nikitaved) [reply aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu:498]: Regarding tests, I fail to see an explicit test for `mul` in sparse. Does OpInfo work with that?
Edit: found `test_basic_ops`...

Comment (nikitaved) [reply aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu:498]: You are right, sorry for the confusion. I retract my words, then we have a choice between 4 and `values.numel() / values.size(0)`. Since `values` is contiguous, we can still "preserve" the previous behavior of `stride(0)` but use a combination of `numel` and `size(0)` instead. What do you think?

Comment (nikitaved) [reply aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu]: I decided to keep the previous behavior, but made sure that `valueSize` is at least `min(1, stride(0))`. This is, I guess, will be safest option.

**Comment by nikitaved** [comment]
Gradcheck failures seem real. Interesting... Works just fine on my machine.
